template-for-repository/proofs/Makefile.common

```diff
@@ -305,23 +305,26 @@ PROOF_SOURCES ?=
 # report, making the proof run appear to "hang".
 CBMC_TIMEOUT ?= 21600
 
-# Proof writers could add function contracts or loop invariants in
-# CBMC proofs. They should follow three distinct processes:
-# 1. Check whether a function contract is sound, where CHECK_FUNCTION_CONTRACTS
-#    contains the list of functions with contracts to be checked.
-# 2. Replace function calls with their correspondent function contracts,
-#    where USE_FUNCTION_CONTRACTS contains the list of functions with
-#    contracts to be replaced. One must check separately whether a function
-#    contract is sound before replacing it.
-# 3. Apply loop invariants during verification, where APPLY_LOOP_INVARIANTS
-#    contains the list of functions with loop invariants to be applied.
-#    This option is currently unavailable.
+# Proof writers could add function contracts or loop contracts in their source
+# code. These contracts are ignored by default. They may be enabled using three
+# distinct flags:
+# 1. To check whether one or more function contracts are sound with respect to
+#    the function implementation, CHECK_FUNCTION_CONTRACTS should be a list of
+#    function names.
+# 2. To replace calls to certain functions with their correspondent function
+#    contracts, USE_FUNCTION_CONTRACTS should be a list of function names.
+#    One must check separately whether a function contract is sound before
+#    replacing it in calling contexts.
+# 3. To use loop contracts (invariants and variants) for unbounded verification,
+#    USE_LOOP_CONTRACTS should be set to "TRUE".
```

Comment:
This text could be very confusing.  Why is there a check and use function contract and no check and use loop contracts.  (I'm going to have trouble getting used to saying loop contract in place of loop invariant, do we really want to do this?)

Reply:
1. Because check+replace is rolled together for loops. We do check initialization and inductiveness for any provided invariant. It's not useful to separate the check+replace for loop contracts because, unlike function contracts, they cannot be reused.
2. Loop invariants are not the only loop contracts. We already have s2n use cases where we at least need assigns clause. Then we have loop variants coming soon for termination proofs. Those are all enabled by the same flag, because they are interlinked.

Reply:
I split this paragraph and now I talk about function contracts and loop contracts separately.